functests/4_latency/latency.go

```diff
@@ -167,6 +167,78 @@ var _ = Describe("[performance] Latency Test", func() {
 			}
 		})
 	})
+
+	Context("with the cyclictest image", func() {
+		BeforeEach(func() {
+			testName = "cyclictest"
+			initMaximumLatencyValue(testName)
+
+			if profile.Spec.CPU.Isolated == nil {
+				Skip(fmt.Sprintf("Skip the cyclictest test, the profile %q does not have isolated CPUs", profile.Name))
+			}
+		})
+
+		It("should succeed", func() {
+			latencyTestPod = getLatencyTestPod(profile, workerRTNode, testName, []string{})
+			createLatencyTestPod(latencyTestPod)
+
+			// verify the maximum latency only when it requested, because this value can be very different
+			// on different systems
+			if maximumLatency == -1 {
+				return
+			}
+
+			latencies := extractLatencyValues("cyclictest", `# Max Latencies:\t*\s*(.*)\s*\t*`, workerRTNode)
+			for _, lat := range strings.Split(latencies, " ") {
+				if lat == "" {
+					continue
+				}
+
+				curr, err := strconv.Atoi(lat)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(curr < maximumLatency).To(BeTrue(), "The current latency %d is bigger than the expected one %d", curr, maximumLatency)
+			}
+		})
+	})
+
+	Context("with the hwlatdetect image", func() {
+		BeforeEach(func() {
+			testName = "hwlatdetect"
+			initMaximumLatencyValue(testName)
+		})
+
+		It("should succeed", func() {
+			// This value should be > than max latency,
+			// in order to prevent the hwlatdetect return with error 1 in case latency value is bigger than expected.
+			// in case latency value is bigger than expected, it will be handled on different flow.
```

Comment:
so should it possibly overridable by users, but with a sane defaults?

Reply:
This is just a sanity test for the tools per se.
The user case is different, he will run the ```cnf-tests``` image directly and provides the parameters as it sees fit.

Reply:
What exactly hard limit should check?

Reply:
TL;DR - A hack to avoid ```hwlatdetect``` from return with an error.
As mentioned in the comment, in case that the actual latency is bigger than the expected one,  ```hwlatdetect``` will return with an error.
This is undesired behavior because we aren't always want to fail the test on anomaly latency values (in case we do, we have a separate check for that [here](https://github.com/openshift-kni/performance-addon-operators/pull/664/files#diff-a11540b25a190b62a4789801be3fa3b70b91847dc903680998aa8a5837550646R252))

So by configuring ```hardlimit``` we are telling ```hwlatdetect``` to ignore the configured latency value, and return with an error only if the latency value is bigger than the ```hardlimit``` value instead (which is incredibly high and not expected).

Reply:
so, you can rewrite the test
```
hardLimit := maximumLatency
if hardLimit == -1 {
    hardlimit := 1000
}

hwlatdetectArgs := []string{
    fmt.Sprintf("-hardlimit=%d", hardlimit),
}

latencyTestPod = getLatencyTestPod(profile, workerRTNode, testName, hwlatdetectArgs)
createLatencyTestPod(latencyTestPod)
```
and you can drop all part related to latency parsing

Reply:
I thought about that, but then we might not be able to distinguish between an error code that returns from ```hwlatdetect``` because of high latency value or some unrelated internal error, right?

Reply:
you can just print the log output at the end of the test to make it more clear, you anyway do not return any viable error in case if the pod failed to start.